src/main/flight/pid.c

```diff
@@ -349,9 +350,12 @@ static void pidApplyRateController(const pidProfile_t *pidProfile, pidState_t *p
 #endif
 }
 
-void updateMagHoldHeading(int16_t heading)
+void updateMagHoldHeading(int16_t heading, uint8_t reset)
```

Comment:
Rather than add a reset parameter, can you add a new function `resetMagHoldHeading` and call that appropriately. Boolean parameters like this make the code difficult to read and understand and can cause errors later down the line.